src/lib/helpers/notifications.ts

```diff
@@ -14,12 +14,12 @@ export type NotificationCoolOffOptions = {
     exponentialBackoffFactor?: number;
 };
 
-const userPreferences = () => get(user).prefs;
+const userPreferences = () => get(user)?.prefs;
 
 const notificationPrefs = (): Record<string, NotificationPrefItem> => {
     const prefs = userPreferences();
     // for some reason, the prefs become array as default or on all clear. let's reset.
-    return Array.isArray(prefs.notificationPrefs) ? {} : prefs.notificationPrefs || {};
+    return prefs && Array.isArray(prefs.notificationPrefs) ? {} : prefs.notificationPrefs || {};
```

Comment:
Isn't this still a problem because if `prefs` is null, the condition will be false and it will try to evaluate `prefs.notificationPrefs`?

Reply:
Yeah you're right. This ternary operator is also a bit hard to read. I've made a new PR to update this:
https://github.com/appwrite/console/pull/1481